### DISCUSSION TITLE
Comment créer une nouvelle branche git ?

### DISCUSSION THREAD

**Discussion by Amis-BASH** [community/community#147661] [Other Feature Feedback, Questions, & Ideas] [answered]
--> Cela fonctionne et permet d'aller sur la branche, sans erreur :
a="nouveaunomdebranche"
git checkout -b $a ; git push --set-upstream origin $a ;

--> Cela ne fonctionne pas, aller sur la branche affiche une erreur fatale.
git checkout -b nom_de_branche 
Cela affiche la nouvelle branche 'nom_de_branche' avec un message d'erreur :

fatal : argument 'origin/nom_de_branche..HEAD' ambigu : révision inconnue ou chemin inexistant. Utilisez '--' pour séparer les chemins des révisions, comme ceci : 'git <commande> [<révision>...] -- [<chemin>...]'

--> Je veux créer une branche et y aller avec " git checkout branche " sans avoir cette erreur.

Quelle est la configuration de git a modifier ?
Pourquoi les projets git normaux ne rencontre pas cette erreur ?

**Comment by imrkaofficial** [comment]
Steps to resolve:

1. Mettre à jour les références distantes : Assurez-vous que votre dépôt local est à jour avec les branches distantes. Vous pouvez le faire avec:

> `git fetch --all `

2. Créez et suivez correctement la branche distante : Si vous créez une nouvelle branche et souhaitez qu'elle suive correctement une branche distante, vous pouvez utiliser cette commande pour définir la branche distante comme "upstream" dès le début :

> `git checkout -b nom_de_branche origin/nom_de_branche`

3.  Sinon, si vous créez simplement la branche et voulez la pousser ensuite, vous pouvez utiliser :

>` git checkout -b nom_de_branche`

> `git push --set-upstream origin nom_de_branche`

4. Vérification de la configuration de votre dépôt local : Parfois, une mauvaise configuration des références locales ou distantes peut provoquer ce type d'erreur. Vous pouvez vérifier la configuration des remotes avec 

> `git remote -v`

5. Cela affichera les URL des remotes pour vous assurer que vous travaillez sur le bon dépôt distant. Si nécessaire, vous pouvez reconfigurer ou ajouter des remotes avec :

> `git remote set-url origin <url_du_dépôt>`

6. Option '--' pour séparer les révisions : Si vous rencontrez toujours l'erreur, vous pouvez essayer d'ajouter l'option -- pour séparer les révisions des chemins de fichiers. Par exemple :

> `git checkout -b nom_de_branche -- .`



Résumé :
Utilisez `git fetch --all` pour mettre à jour les références distantes.
Créez la branche et définissez correctement le suivi avec `git push --set-upstream origin nom_de_branche`.
Assurez-vous que la configuration des remotes est correcte et que les branches distantes sont à jour.

**Comment by Amis-BASH** [comment]
Bonjour et merci de votre réponse.

`git checkout -b nom_de_branche origin/nom_de_branche`
fatal : 'origin/nom_de_branche' n'est pas un commit et une branche 'nom_de_branche' ne peut pas en être créée depuis

Ces deux commandes fonctionnent sans afficher d'erreurs de branche :
```
git checkout -b nom_de_branche
git push --set-upstream origin nom_de_branche
```

**Comment by Amis-BASH** [comment, marked as answer]
Mon script PS1 était en erreur.
La fonction Bash pour PS1 ne recherchait pas correctement les références dans HEAD.

Je partager cette fonction mise à jour :
https://github.com/Amis-BASH/partage-git-public/issues/1

Cela pourrait servir ...

**Comment by shinybrightstar** [comment]
Hi @Amis-BASH , thanks for participating!

Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting to ensure discussions are adhering to our [Code of Conduct](https://github.com/community/community/blob/920a928ffe3ae4863b64a4ea26917046e11f5c1a/CODE_OF_CONDUCT.md). 

We’ll be locking this post for now, but please feel encouraged to create a new discussion in English if you still need support.
